### ISSUE TITLE
 [Whitelist] Sonic Mania (584400)

### ISSUE THREAD

**Issue by apIncubout** [ValveSoftware/Proton#105]
Work correct on Linux mint 18.3

Not issues from the moment.

Hardware: i5 7400 +  Intel Graphics card 
Linux Mint 18.3

**Comment by mercuriete** [comment]
Work correct on Gentoo

uname -a
Linux localhost.localdomain 4.14.63-gentoo-r1 1 SMP PREEMPT Sun Aug 19 04:55:38 CEST 2018 x86_64 Intel(R) Core(TM) i7-4720HQ CPU @ 2.60GHz GenuineIntel GNU/Linux

DRI_PRIME=1 glxinfo |grep "OpenGL version string"
OpenGL version string: 4.4 (Compatibility Profile) Mesa 18.2.0-rc3

DRI_PRIME=1 glxinfo |grep "renderer string"
OpenGL renderer string: AMD Radeon HD 8800M Series (CAPE VERDE, DRM 3.19.0, 4.14.63-gentoo-r1, LLVM 6.0.1)

**Comment by puffnfresh** [comment]
No issues on NixOS!

**Comment by SCheeseman** [comment]
No issues on Ubuntu 18.04 /w GTX 1080.

**Comment by bruno-fs** [comment]
failed to run on Ubuntu 18.04 with GTX 1050

```
$ uname -a
Linux firelord 4.15.0-32-generic #35-Ubuntu SMP Fri Aug 10 17:58:07 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

$ glxinfo |grep "OpenGL version string"
OpenGL version string: 4.6.0 NVIDIA 396.51

$  glxinfo |grep "renderer string"
OpenGL renderer string: GeForce GTX 1050/PCIe/SSE2
```
![screenshot_20180823_054118](https://user-images.githubusercontent.com/6073215/44514777-57e99b00-a697-11e8-8d1f-bf9955d9f9c4.png)


EDIT: today I got a new version of nvidia-driver and now it is working! 
```bash
$ glxinfo | grep "OpenGL version string"
OpenGL version string: 4.6.0 NVIDIA 396.54
```

**Comment by daniel-j** [comment]
I have the same issue with the server is not reachable error message. Arch Linux with GTX 980 Ti

**Comment by parkerlreed** [comment]
Working here on Arch with AMDGPU. Was actually surpised since by default Mania needs xact to fix choppy audio. Is this already included as a lib/one of the patches?

**Comment by aqxa1** [comment]
@parkerlreed 
I had a look at the prefix, and the xaudio2_[0-7] files appear to be the native ones, since their filesizes are around 500kb each, whereas the built in dlls are just text files with a few lines of code. So yes, it looks like xact is already included.

**Comment by winghugs** [comment]
Confirming that I get "server is not reachable" as well. 
RX 480
Ryzen 5 1600
Mesa 18.1
LLVM 6.0

**Comment by PhonicUK** [comment]
Working flawlessly here, Ubuntu 18.04, Ryzen 7 / Vega 10

**Comment by dominichayesferen** [comment]
Working without any flaws here, heck it even installs the Plus DLC fine right from the game, in Proton.

**Comment by flarn2006** [comment]
I'm getting the choppy audio issue here. I tried installing xact (using winetricks, setting WINEPREFIX to the game's compatdata folder) but that didn't fix it.

**EDIT:** Saw someone suggested changing the emulated Windows version to XP, and that did it for me. Thanks!

**Comment by flibitijibibo** [comment]
@kisak-valve Requesting an XAudio2 label for this game...

**Comment by diegogangl** [comment]
Works flawlessly. Finished the game + DLC. Put about 12 hours so far and no issues.

- Fedora 29
- Kernel 4.18.16
- NVIDIA 410.66

**Comment by Thesaurus-Kit** [comment]
The game do not detect Dualshock 4 on the first launch after Steam start. If I close the game using keyboard and launch it again Dualshock works properly until next Steam restart.

Proton version 3.16-4 beta
OS Fedora 29

**Comment by Thesaurus-Kit** [comment]
It seems that enabling of "Playstation configuration support" in BigPicture settings solved the Dualshock 4 issue in previous comment.

**Comment by mimattr** [comment]
# Whitelist Request
- Name of the game to be whitelisted: **Sonic Mania**
- Steam AppID of the game: **584400**

## System Information
- GPU: **GTX 970**
- Driver/LLVM version: **NVIDIA 415.25**
- Distro version: **Ubuntu 16.04.5 LTS**
- [Link to full system information report as Gist](https://gist.github.com/mimattr/b0b7f46366e31e254460b9593ce87e51)
- Proton version: **3.16-6 Beta**

## I confirm:
- [x] that pressing the `Play` button in the Steam client is sufficient.
- [ ] that [runtime config options](https://github.com/ValveSoftware/Proton#runtime-config-options) are necessary to run the game.
- [ ] that no workarounds other than PROTON_* env variables are necessary.

## Issues
- [ ] I haven't experienced any issues.
- [ ] There are no issues left open for this game.
- [x] Although I consider the gaming experience equal to Windows there are remaining issues:
  - Application appears to work as expected but detects an [incorrect display resolution](https://gist.github.com/mimattr/4c5857150cc580527b5240590fdf58e4) during launch process (not fullscreen).

**Comment by serebit** [comment]
No audio issues, but the game flat-out doesn't recognize my PS4 controller, while other games do.

**Comment by mercuriete** [comment]
@serebit 
Not sure why, but I can use it if I disable steam controller option and use it as a normal controller.
my ps4 controller never works on any proton title if I opted in for ps4 steam controller feature.


From my point of view sonic is working perfect.

**Comment by serebit** [comment]
Works fine with my PS4 controller with 4.11-4.

**Comment by skeevy420** [comment]
Manjaro Linux 5.3.1-6.1-tkg-bmq
MSI RX 580
Mesa 19.3.0_devel.115749.336b021d36f-1
Tested Proton 4.11-6; proton-tkg with dxvk master; wine-staging 4.17
LLVM 10.0.0_r327586.23aac95a32c-1
amdvlk 2019.Q3.6-2
Tested with FSYNC on and off

Just got this game with this month's HumbleBundle and I was unable to get it to run until I set `VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/amd_icd32.json %command%` or `VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/radeon_icd.i686.json %command%` under the launch options.

The only error in my logs was: 
```
terminate called after throwing an instance of 'dxvk::DxvkError'
pid 37033 != 37032, skipping destruction (fork without exec?)
```

Using exiftool on the binary showed it to be a 32bit executable which gave me the idea to try to use a 32bit vulkan implementation which works great and makes it perfectly playable for me.

Works fine with my PS4 controller (Bluetooth).

EDIT:

Realized after posting this that I set a 64bit vulkan implementation globally upon boot.  While I have to leave the house right now, I'll test this game when I get home in another hour or two without radeon_icd.x86_64 set globally on boot and see if that fixes my issue and turns this from a Proton/Wine issue to my setup being the issue.

UPDATE:

It was because of my globally set 64bit Vulkan ICD.  Removed it and all my launch options and it works just fine.

**Comment by Octelly** [comment]
Doesn't even launch on Linux Mint 19.1 on Thinkpad T430 (Intel iGPU) with Proton 4.11-12

**Comment by kisak-valve** [comment]
Hello @Johnystar, please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options and drag and drop the generated $HOME/steam-$APPID.log into the comment box. Also, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by Octelly** [comment]
I have tried switching to Proton 5.0-2 in the meantime with no luck.

[Log + system information here](https://gist.github.com/Johnystar/0e22fcf763f90211e0a41549b1761d92)

**Comment by Octelly** [comment]
A Steam update has just been released, so [here's a new system information report](https://gist.github.com/Johnystar/738eecfb30785c1c94a332ae04d7c03f)

**Comment by kisak-valve** [comment]
`info:  Required Vulkan extension VK_KHR_surface not supported` -> `terminate called after throwing an instance of 'dxvk::DxvkError'` is the line of interest from your log. Please check if any vulkan application like `vulkaninfo` from `vulkan-utils` works.

Mesa also has an incomplete vulkan implementation for your Ivybridge generation hardware. You could try updating to a newer mesa version using a PPA like [oibaf](https://launchpad.net/~oibaf/+archive/ubuntu/graphics-drivers) or [kisak-mesa](https://launchpad.net/~kisak/+archive/ubuntu/kisak-mesa).

Alternatively, you could try adding `PROTON_USE_WINED3D=1 %command%` to the game's launch options to try wine's DirectX to OpenGL render path, but this is unsupported and is expected to come with a performance penalty.

**Comment by Octelly** [comment]
After installing `vulkan-utils` onto my system, I got the following output from ``vulkaninfo``:

```
===========
VULKAN INFO
===========

Vulkan Instance Version: 1.1.70

Cannot create Vulkan instance.
/build/vulkan-UL09PJ/vulkan-1.1.70+dfsg1/demos/vulkaninfo.c:768: failed with VK_ERROR_INCOMPATIBLE_DRIVER
```

Incompatible driver? Sounds like the Mesa incompatibility you mentioned... I added the **oibaf** repository and ran apt update & upgrade to get the latest versions of all packages on my system. That didn't really do much and I'm still getting the same ``vulkaninfo`` output as before.

Enabling Wine 3D or ``PROTON_USE_WINED3D=1`` worked alright, but the game did indeed use more system resources than I'd usually expect and when I try switching to fullscreen it crashes, which is a significant problem considering the game without it covers a quarter of the laptop's screen at best.

**Comment by kisak-valve** [comment]
You can also double check that you have the `mesa-vulkan-drivers` and `mesa-vulkan-drivers:i386` packages installed with something like `apt policy mesa-vulkan-drivers mesa-vulkan-drivers:i386`.

**Comment by Octelly** [comment]
Yep, ``Installed: (none)``. I guess that explains a lot. After installing the necessary drivers, the game now runs as expected including fullscreen and everything.

Thanks for your help and time :D

**Comment by Ruedii** [comment]
It used to have audio issues for me but works fine now.

System Info:
https://gist.github.com/Ruedii/a1dacec22e0649421574050423a5529c

**Comment by K4rakara** [comment]
Audio cuts out at consistent intervals, producing a "crackly" sound on Arch linux 5.10.7-arch1-1

**Comment by sonic2kk** [comment]
Audio issue appears to be fixed though I have never experienced it on any of the hardware I've used. Tested with Proton 6.3-4 and using Arch Linux 5.12.5, but never experienced any audio issues with any versions of Proton I tried (3.16-X, 4.2-X, 4.11-X, 5.0-X, 5.13-X. 6.3-X).

**Comment by kisak-valve** [comment]
Sonic Mania fails to work unless you use Proton 3.16

Issue transferred from https://github.com/ValveSoftware/Proton/issues/5011.
@coreybruce posted on 2021-07-26T05:06:28:

# Compatibility Report
- Name of the game with compatibility issues:
- Steam AppID of the game: 584400

## System Information
https://gist.github.com/coreybruce/dcd2bec911594c9212d351ac2e7a3f98

- Proton version: 6.3-5

## I confirm:
- [x ] that I haven't found an existing compatibility report for this game.
- [ x] that I have checked whether there are updates for my system available.

Log:
https://gist.github.com/coreybruce/6373473b1bc8bc076e2b7130b6ec0ae4

## Symptoms 
If you use any version newer than proton 3.16 the game will fail to launch

## Reproduction
Launch game with latest version of proton, than switch back to proton 3.16 and you will see the game will only work on that version.

**Comment by kisak-valve** [comment]
Hello @coreybruce, your Proton log indicates a general failure to initialize Vulkan on your system, the same as #4979. The summary of your hardware doesn't really give any insight into how healthy the Vulkan render paths are, like we see in the extended diagnostics section of Steam's system information. (`Steam` -> `Help` -> `System Information`)

Most likely an issue with your system, and not the game / Proton.

**Comment by mercuriete** [comment]
I just installed sonic mania on my setup and it works on proton 6.3-5

I didn't do anything different to click on play button and it works flawlessly.

It works for me.

PS: the problem that I commented on this issue before about controller not working was a problem that I had on my side.
now sonic works flawlessly with ps4 controller without any issue.

**Comment by coreybruce** [comment]
> Hello @coreybruce, your Proton log indicates a general failure to initialize Vulkan on your system, the same as #4979. The summary of your hardware doesn't really give any insight into how healthy the Vulkan render paths are, like we see in the extended diagnostics section of Steam's system information. (`Steam` -> `Help` -> `System Information`)
> 
> Most likely an issue with your system, and not the game / Proton.

<details><summary>Here is my Steam log:</summary>

```
Computer Information:
    Manufacturer:  Unknown
    Model:  Unknown
    Form Factor: Desktop
    No Touch Input Detected

Processor Information:
    CPU Vendor:  GenuineIntel
    CPU Brand:  Intel(R) Core(TM) i5-8600K CPU @ 3.60GHz
    CPU Family:  0x6
    CPU Model:  0x9e
    CPU Stepping:  0xa
    CPU Type:  0x0
    Speed:  4300 Mhz
    6 logical processors
    6 physical processors
    HyperThreading:  Unsupported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Supported
    AES:  Supported
    AVX:  Supported
    AVX2:  Unsupported
    AVX512F:  Unsupported
    AVX512PF:  Unsupported
    AVX512ER:  Unsupported
    AVX512CD:  Unsupported
    AVX512VNNI:  Unsupported
    SHA:  Unsupported
    CMPXCHG16B:  Supported
    LAHF/SAHF:  Supported
    PrefetchW:  Unsupported

Operating System Version:
    "Manjaro Linux" (64 bit)
    Kernel Name:  Linux
    Kernel Version:  5.10.52-1-MANJARO
    X Server Vendor:  The X.Org Foundation
    X Server Release:  12011000
    X Window Manager:  Xfwm4
    Steam Runtime Version:  steam-runtime_0.20210630.0

Video Card:
    Driver:  Mesa/X.org llvmpipe (LLVM 12.0.1, 256 bits)
    Driver Version:  3.1 Mesa 21.1.5
    OpenGL Version: 3.1
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 60 Hz
    VendorID:  0x10de
    DeviceID:  0x1c03
    Revision Not Detected
    Number of Monitors:  1
    Number of Logical Video Cards:  1
    Primary Display Resolution:  1920 x 1080
    Desktop Resolution: 1920 x 1080
    Primary Display Size: 20.91" x 11.77" (23.98" diag)
                                            53.1cm x 29.9cm (60.9cm diag)
    Primary Bus: PCI Express 16x
    Primary VRAM Not Detected
    Supported MSAA Modes:  2x 4x 8x 16x 

Sound card:
    Audio device: Realtek ALC887-VD

Memory:
    RAM:  15947 Mb

VR Hardware:
    VR Headset: None detected

Miscellaneous:
    UI Language:  English
    LANG:  en_AU.utf8
    Total Hard Disk Space Available:  102102 Mb
    Largest Free Hard Disk Block:  9895 Mb

Storage:
    Number of SSDs: 0
    SSD sizes: %s1
    Number of HDDs: 0
    HDD sizes: %s1
```

</details>

**Comment by kisak-valve** [comment]
> Driver:  Mesa/X.org llvmpipe (LLVM 12.0.1, 256 bits)

This is mesa's faster CPU renderer, it means something's misconfigured and you're not using your nVidia GPU to render (at least for OpenGL).

**Comment by coreybruce** [comment]
Yep confirm it was a issue on my end and has been fixed, the game now runs with the latest proton just fine. Sorry about that.

**Comment by kentosama** [comment]
# Compatibility Report
- Name of the game with compatibility issues: Sonic Mania
- Steam AppID of the game: 584400 

## System Information
- GPU: Radeon 280x
- Driver/LLVM version: Mesa 22.1.5
- Kernel version: 5.18.16-arch1-1
- Link to full system information report as [Gist](https://gist.github.com/kentosama/5a56a8c306bbca36a50b8985148f5783):
- Proton version: 3.7/8 and experimental

## I confirm:
- [ x ] that I have checked whether there are updates for my system available.

## Symptoms 
Sonic Mania does not launch

## Reproduction
Run the game on ArchLinux

**Comment by kisak-valve** [comment]
Hello @kentosama, these look like the lines of interest from your log:
```
err:module:load_builtin_dll failed to load .so lib for builtin L"WLDAP32.dll": libcrypt.so.1: cannot open shared object file: No such file or directory
warn:module:load_dll Failed to load module L"WLDAP32.dll"; status=c000007a
err:module:import_dll Loading library WLDAP32.dll (which is needed by L"Z:\\home\\kentosama\\.local\\share\\Steam\\steamapps\\common\\Sonic Mania\\SonicMania.exe") failed (error c000007a).
[...]
err:module:attach_dlls Importing dlls for L"Z:\\home\\kentosama\\.local\\share\\Steam\\steamapps\\common\\Sonic Mania\\SonicMania.exe" failed, status c0000135
```
Glibc used to provide libcrypt.so.1, but no longer does. On Arch, the `libxcrypt-compat` and `lib32-libxcrypt-compat` system packages should now provide that library.

Can you share a Proton log from trying the game with Proton 5.13 or newer?
